openhcl/underhill_mem/src/lib.rs

```diff
@@ -611,20 +620,28 @@ mod mapping {
         }
 
         /// Query the current permissions for a vtl on a page.
-        fn vtl_permissions(&self, vtl: Vtl, gpa: u64) -> GpaVtlPermissions {
+        fn vtl_permissions(
+            &self,
+            vtl: Vtl,
+            gpa: u64,
+        ) -> Result<GpaVtlPermissions, QueryVtlPermissionsError> {
             match self.isolation {
                 IsolationType::None | IsolationType::Vbs => unimplemented!(),
                 IsolationType::Snp => {
                     // TODO CVM GUEST VSM: track the permissions directly in
-                    // underhill. For now, use rmpquery.
+                    // underhill. For now, use rmpquery--but note this is only
+                    // supported on Genoa+.
                     assert!(self.isolation == IsolationType::Snp);
```

Comment:
Is this assert useful? I think it's vacuously true after you've matched `self.isolation` to `IsolationType::Snp`.

Reply:
oops thought I deleted that